doc/loganalyzer_to_pytest.md

```diff
@@ -0,0 +1,103 @@
+- Perform analysis of downloaded syslog and return the result
+
+Current loganalyzer implementation description:
+https://github.com/Azure/SONiC/wiki/LogAnalyzer
+
+#### Development
+Module "loganalyzer.py" with class "Loganalyzer".
+
+"Loganalyzer" class interface:
+- __init__(ansible_host: ansible_host, marker_prefix, dut_run_dir="/tmp")
+- load_common_config() - Clear previous configured match, expect and ignore. Load regular expressions from common configuration files: match, expect and ignore which are located in some configuration directory or with current module. Save loaded configuration to the self.match_regex, self.expect_regex and self.ignore_regex attributes.
+- parse_regexp_file(file_path) - Read and parse regular expressions from specified file. Return list of strings of defined regular expressions.
+- run_cmd(callback, *args, **kwargs) - Call function and analyze DUT syslog during function execution. Return the same result as "analyze" function.
+- init() - Add start marker to the DUT syslog. Generated marker format: marker_prefix + "%Y-%m-%d-%H:%M:%S"
+- analyze() - Extract syslog based on the start marker and copy it to ansible host. Analyze extracted syslog file localy. Return python dictionary object.
+Return example:
+{"counters": {"match": 1, "expected_match": 0, "expected_missing_match": 0},
+ "match_files": {"/tmp/syslog": {"match": 0, "expected_match": 32},
+					"/tmp/syslog1": {"match": 0, "expected_match": 15}
+						},
+ "match_messages": {"/tmp/syslog1": ["Message 1", "Message 2", "Message n"],
+						 "/tmp/syslog2": ["Message 1", "Message 2", "Message n"]
+						 }
+}
+- save_full_log(dest) - Download extracted DUT syslog (/tmp/syslog) to the Ansible host folder specified in 'dest' input parameter.
+
+Attributes:
+- match_regex - list of regular expression strings to match
+- expect_regex - list of regular expression strings to expect
+- ignore_regex - list of regular expression strings to ignore
+
+Usage example of loganalyzer functionality just to show how to use loganalyzer interface.
+
+```
+def some_function(x, y=10, z=0):
+    return x + y
+
+def test_loganalyzer_functionality(localhost, ansible_adhoc, testbed):
+    """
+    @summary: Example of loganalyzer usage
+    """
+
+    hostname = testbed['dut']
+    ans_host = ansible_host(ansible_adhoc, hostname)
+
+    log = LogAnalyzer(ansible_host=ans_host, marker_prefix="test_loganalyzer")
+
+    # Read existed common regular expressions located with legacy loganalyzer module
+    log.load_common_config()
+
+    # Add start marker to the DUT syslog
+    log.init()
+
+    # Emulate that new error messages appears in the syslog
+    time.sleep(1)
+    ans_host.command("echo '---------- ERR: error text --------------' >> /var/log/syslog")
+    ans_host.command("echo '---------- Kernel Error: error text --------------' >> /var/log/syslog")
+    time.sleep(2)
+    ans_host.command("echo '---------- Interface Error: error text --------------' >> /var/log/syslog")

```

Comment:
can we use `logger` command?

Reply:
This ans_host.command performs shell commands on the DUT.
For display messages to the display yes we can use logger. But for writing something to the DUT files we cant use logger because this file is not running on the DUT.